From c1d1281b1ce7adc3075a902a0f6fadfbcc53e882 Mon Sep 17 00:00:00 2001
From: Sebastian Fricke <sebastian.fricke.linux@gmail.com>
Date: Sun, 10 Jan 2021 07:22:32 +0100
Subject: [PATCH] arm64:dts: Add OV13850 to the device tree

The OV13850 camera sensor device tree addition to the I2C1 node
was ported from the BSP kernel from friendlyElec.

Define the configuration for the regulators used by the OV13850
camera sensor. The values are taken from the datasheet. See:
(http://www.t-firefly.com/download/firefly-rk3288/peripherals/Sensor_OV13850-G04A_OmniVision_Specification(V1.1).pdf#page=26)

Increase the temperature limits (ported from rockchip64-dev armbian tree and embedded
into this patch)

Signed-off-by: Sebastian Fricke <sebastian.fricke.linux@gmail.com>
---
 .../boot/dts/rockchip/rk3399-nanopc-t4.dts    | 92 +++++++++++++++++++
 arch/arm64/boot/dts/rockchip/rk3399.dtsi      |  6 +-
 2 files changed, 95 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts b/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts
index e0d75617bb7e..8b6301b0dd8f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts
@@ -32,6 +32,30 @@ vcc5v0_host0: vcc5v0-host0 {
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	ov13850_avdd_2p8v: 2p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov13850_avdd";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		regulator-always-on;
+	};
+
+	ov13850_dovdd_1p8v: 1p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov13850_dovdd";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+	};
+
+	ov13850_dvdd_1p2v: 1p2v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov13850_dvdd";
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+		regulator-always-on;
+	};
+
 	adc-keys {
 		compatible = "adc-keys";
 		io-channels = <&saradc 1>;
@@ -99,6 +123,34 @@ &pcie0 {
 	vpcie3v3-supply = <&vcc3v3_sys>;
 };
 
+
+&i2c1 {
+	status = "okay";
+	ov13850p0: ov13850@10 {
+		compatible = "ovti,ov13850";
+		status = "okay";
+		reg = <0x10>;
+		clocks = <&cru 0x89>;
+		clock-names = "xvclk";
+		avdd-supply = <&ov13850_avdd_2p8v>;
+		dovdd-supply = <&ov13850_dovdd_1p8v>;
+		dvdd-supply = <&ov13850_dvdd_1p2v>;
+
+		reset-gpios = <&gpio2 27 0>;
+		pwdn-gpios = <&gpio2 28 0>;
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
+		pinctrl-0 = <&cam0_default_pins &cif_clkout_a>;
+		pinctrl-1 = <&cam0_default_pins>;
+
+		port {
+			ucam_out0b: endpoint {
+				remote-endpoint = <&mipi_in_ucam0b>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+};
+
 &pinctrl {
 	ir {
 		ir_rx: ir-rx {
@@ -106,6 +158,46 @@ ir_rx: ir-rx {
 			rockchip,pins = <0 RK_PA6 1 &pcfg_pull_none>;
 		};
 	};
+
+	cam_pins {
+		cif_clkout_a: cif-clkout-a {
+			rockchip,pins = <2 11 3 &pcfg_pull_none>;
+		};
+
+		cif_clkout_a_sleep: cif-clkout-a-sleep {
+			rockchip,pins = <2 11 0 &pcfg_pull_none>;
+		};
+
+		cam0_default_pins: cam0-default-pins {
+			rockchip,pins = <2 28 0 &pcfg_pull_down>, <2 27 0 &pcfg_pull_none>;
+		};
+
+		cam1_default_pins: cam1-default-pins {
+			rockchip,pins = <0 12 0 &pcfg_pull_down>, <0  8 0 &pcfg_pull_none>;
+		};
+	};
+};
+
+&mipi_dphy_rx0 {
+	status = "okay";
+};
+
+&isp0_mmu {
+	status = "okay";
+};
+
+&isp0 {
+	status = "okay";
+
+	ports {
+		port@0 {
+			mipi_in_ucam0b: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&ucam_out0b>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
 };
 
 &sdhci {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index f5dee5f447bb..dff20b17874d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -772,17 +772,17 @@ cpu_thermal: cpu {
 
 			trips {
 				cpu_alert0: cpu_alert0 {
-					temperature = <70000>;
+					temperature = <85000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
 				cpu_alert1: cpu_alert1 {
-					temperature = <75000>;
+					temperature = <95000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
 				cpu_crit: cpu_crit {
-					temperature = <95000>;
+					temperature = <100000>;
 					hysteresis = <2000>;
 					type = "critical";
 				};
-- 
2.25.1

